### PR TITLE
Add mentoring rep-gate

### PR DESCRIPTION
**Pull request by dem4ron**
Screenshot 2023-03-13 at 17 27 46

### CODE REVIEW

File: app/views/mentoring/external/show.html.haml [hunk 1]
```diff
@@ -11,7 +12,10 @@
         Some of our best mentors worried too.
 
       - if user_signed_in?
-        = ReactComponents::Mentoring::TryMentoringButton.new.to_s
+        - if current_user.reputation >= 20
+          = ReactComponents::Mentoring::TryMentoringButton.new.to_s
```
Comment (iHiD) [inline app/views/mentoring/external/show.html.haml]:
```suggestion
          = ReactComponents::Mentoring::TryMentoringButton.new
```

You shouldn't need `to_s`. It should get automatically called (I know this was in the bit before - maybe you can check for other places to remove it to?)

Comment (dem4ron) [reply app/views/mentoring/external/show.html.haml]: ok!

File: app/views/mentoring/external/show.html.haml [hunk 2]
```diff
@@ -11,7 +11,7 @@
         Some of our best mentors worried too.
 
       - if user_signed_in?
-        - if current_user.reputation >= 20
+        - if current_user.reputation >= 20 || current_user.mentor?
```
Comment (iHiD) [inline app/views/mentoring/external/show.html.haml]: If they current user is a mentor they should never see this page in the first place.

File: app/commands/user/become_mentor.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class User::BecomeMentor
   initialize_with :user, :track_slugs
 
   def call
-    return if user.mentor?
+    return if user.mentor? || user.reputation < 20
```
Comment (iHiD) [inline app/commands/user/become_mentor.rb]: Let's put this 20 into a constant (maybe in `User` called `MIN_REP_TO_MENTOR`) and then update everywhere that's 20 in this PR to `User::MIN_REP_TO_MENTOR`.

**Pull request merged**